kernel/Makefile

```diff
@@ -0,0 +1,196 @@
+# OS X kext makefile
+VERSION:=$(shell git describe --tags HEAD --always)
+VERSION_DEFINE:=-DVERSION="\"$(VERSION)\""
+
+CC:=clang
+CXX:=clang++
+
+# Directories
+INCLUDE_DIR:=include/
+BUILD_DIR:=build/
+SOURCE_DIR:=src/
+KEXT:=osquery.kext
+CONTENTS_DIR:=$(BUILD_DIR)$(KEXT)/Contents/
+OUTPUT_DIR:=$(CONTENTS_DIR)MacOS/
+RUNDIR:=/tmp/
+USER_DIR:=user/
+
+INCLUDES:=$(wildcard $(INCLUDE_DIR)*.h)
+INCLUDE:=-I$(INCLUDE_DIR) -I/System/Library/Frameworks/Kernel.framework/Headers
```

Comment:
Does this exist on every system? Should we use:

```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX[INSERT_VERSION].sdk/System/Library/Frameworks/Kernel.framework/Headers
```

Maybe have make logic that determines the platform, `../tools/provision.sh get_platform` and then set the SDK accordingly. So we only build 10.10 on a 10.10 build host?